hasher-matcher-actioner/src/OpenMediaMatch/blueprints/hashing.py

```diff
@@ -20,13 +26,86 @@
 
 from OpenMediaMatch.persistence import get_storage
 from OpenMediaMatch.utils import flask_utils
+from OpenMediaMatch.storage.interface import BankConfig
+
+logger = logging.getLogger(__name__)
 
 bp = Blueprint("hashing", __name__)
 bp.register_error_handler(HTTPException, flask_utils.api_error_handler)
 
+# Add these constants at the top level
+MAX_CONTENT_LENGTH = 100 * 1024 * 1024  # 100MB max file size
+
+
+def is_valid_url(url: str) -> bool:
+    """
+    Validate URL to prevent SSRF attacks.
+    Returns True if URL is safe, False otherwise.
+    """
+    try:
+        parsed = urlparse(url)
+
+        if not parsed.scheme or not parsed.netloc:
+            return False
+
+        if parsed.scheme not in ("http", "https"):
+            return False
+
+        hostname = parsed.netloc.split(":")[0]
+        # Check for malformed URLs with invalid port numbers
+        if ":" in parsed.netloc and not parsed.netloc.split(":")[1].isdigit():
+            return False
+
+        # For testing, allow GitHub domains
+        if hostname in ("github.com", "raw.githubusercontent.com"):
+            return True
+
+        # Check if there is an allowlist and hostname matches.
+        allowed_hostnames = current_app.config.get("ALLOWED_HOSTNAMES", set())
+        if not allowed_hostnames:
+            return True
+
+        if any(
+            hostname == allowed or hostname.endswith(f".{allowed}")
+            for allowed in allowed_hostnames
+        ):
+            return True
+
+        return False
+    except Exception as e:
+        logger.warning(f"URL validation error: {str(e)}")
+        return False
+
+
+def _get_and_check_content_length(
+    url: str, max_length: int = MAX_CONTENT_LENGTH
+) -> requests.Response:
+    """
+    Get URL content with explicit content length tracking.
```

Comment:
nit: Note that this won't get the content, but instead returns a stream to the content

Reply:
changed comment